### PR TITLE
[teams] inactivate Leave Team action for last team owner

### LINKED ISSUE #4623: Prevent last owner from leaving the team
Currently in Teams & Projects:
- Owners can remove Members form the team
- Anyone can leave a team

However, if you're the last Owner of the team and you leave, the team has no owner anymore.

Also, if you're the last person on the team, the team will become empty (and unless someone still has a working invite link, the team name/slug is "burned").

From https://github.com/gitpod-io/gitpod/pull/4571#discussion_r658750565:

> Some of the possible solutions here:
> 1. Keep the _Leave_ option as is and on click trigger a toast notification with an error message explaining that you cannot leave because you are the only remaining owner of the team. Needs: https://github.com/gitpod-io/gitpod/issues/3530.
> 2. Keep the _Leave_ option as a disabled option (`red-300`) and show a tooltip on hover explaining why you cannot perform this action. See screenshot A below.
> 3. Disable the more actions button (`gray-400`) and show a tooltip on hover. See screenshot B below.
> 
> | A. Disabled dropdown option | B. Disabled dropdown button |
> |-|-|
> | Screenshot 2021-06-25 at 4 06 33 PM |  Screenshot 2021-06-25 at 4 08 04 PM |

### PR DESCRIPTION
**Pull request by AlexTugarev**
Fixes #4623

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#4623*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/teams/Members.tsx [hunk 1]
```diff
@@ -46,6 +47,10 @@ export default function() {
         })();
     }, [ team ]);
 
+    useEffect(() => {
+        setLeaveTeamEnabled(members.length > 1);
```
Comment (JanKoehnlein) [inline components/dashboard/src/teams/Members.tsx:51]: Shouldn't it be filtered by team owners?

Comment (AlexTugarev) [reply components/dashboard/src/teams/Members.tsx]: absolutely! just updated

**Comment by JanKoehnlein** [comment]
Doesn't work for me:
1) Create a team (thus only yourself as a member and owner)
2) Go to team page and delete yourself
Gitpod allows that

**Comment by AlexTugarev** [comment]
@JanKoehnlein, thanks for testing! Sorry, I just confused `active` state with enablement.

**Comment by AlexTugarev** [comment]
Screen Shot 2021-11-01 at 10 55 33

This should be available with next update.

File: components/dashboard/src/teams/Members.tsx [hunk 2]
```diff
@@ -179,9 +186,9 @@ export default function() {
                             <span className="flex-grow" />
                             <ItemFieldContextMenu menuEntries={m.userId === user?.id
                                 ? [{
-                                    title: 'Leave Team',
-                                    customFontStyle: 'text-red-600 dark:text-red-400 hover:text-red-800 dark:hover:text-red-300',
-                                    onClick: () => removeTeamMember(m.userId)
+                                    title: leaveTeamEnabled ? 'Leave Team' : 'Remaining owner',
+                                    customFontStyle: leaveTeamEnabled ? 'text-red-600 dark:text-red-400 hover:text-red-800 dark:hover:text-red-300' : 'text-gray-400 dark:text-gray-200',
```
Comment (laushinka) [inline components/dashboard/src/teams/Members.tsx:190]:
```suggestion
                                    customFontStyle: leaveTeamEnabled ? 'text-red-600 dark:text-red-400 hover:text-red-800 dark:hover:text-red-300' : 'text-gray-400 dark:text-gray-200 cursor-not-allowed',
```

How about also adapting the cursor to show that there is no action? E.g. `cursor-not-allowed` or `cursor-default`.

Comment (AlexTugarev) [reply components/dashboard/src/teams/Members.tsx:190]: Given that the label is changed to `Remaining owner`, I don't think it's necessary to change the cursor. Also, there is no other example in the dashboard to provide feedback using the "not-allowed" cursor.

Comment (geropl) [reply components/dashboard/src/teams/Members.tsx:190]: Would it make sense to have some server-side validation for this as well?

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: b4c1c8a0076591d28fb42b80cd49323b09f0c5a2</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6477#issuecomment-957323173" title="LGTM">JanKoehnlein</a>*

Associated issue: *#4623*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**